connection_test.go

```diff
@@ -765,3 +767,42 @@ func TestConnectTimeout(t *testing.T) {
 		close(testComplete)
 	})
 }
+
+func TestTosPriority(t *testing.T) {
+	ctx, cancel := NewContext(time.Second)
+	defer cancel()
+
+	opts := testutils.NewOpts().SetServiceName("s1").SetTosPriority(tos.Lowdelay)
+	testutils.WithTestServer(t, opts, func(ts *testutils.TestServer) {
+		ts.Register(raw.Wrap(newTestHandler(t)), "echo")
+
+		outbound, err := ts.Server().BeginCall(ctx, ts.HostPort(), "s1", "echo", nil)
+		require.NoError(t, err, "BeginCall failed")
+
+		_, outboundNetConn := OutboundConnection(outbound)
+		connTosPriority, err := IsTosPriority(outboundNetConn, tos.Lowdelay)
+		require.NoError(t, err, "Checking TOS priority failed")
+		assert.Equal(t, connTosPriority, true)
+		_, _, _, err = raw.WriteArgs(outbound, []byte("arg2"), []byte("arg3"))
+		require.NoError(t, err, "Failed to write to outbound conn")
+	})
+}
+
+func TestUnmarshalTosPriority(t *testing.T) {
+	expected := tos.Lowdelay
+
+	var TosBit tos.ToS
+	err := TosBit.UnmarshalText([]byte("Lowdelay"))
+	if assert.NoError(t, err, "JSON unmarshal failed") {
+		assert.Equal(t, expected, TosBit, "JSON config mismatch")
+	}
+}
+
+func TestMarshalTosPriority(t *testing.T) {
+	expected := "\"Lowdelay\""
```

Comment:
can avoid the escaping (and i prefer to make these const in a simple test like this)

```go
const expected =`"Lowdelay"`
```